src/YoutubeAtom.tsx

```diff
@@ -161,273 +45,86 @@ export const YoutubeAtom = ({
     eventEmitters,
     pillar,
 }: Props): JSX.Element => {
-    const [iframeSrc, setIframeSrc] = useState<string | undefined>(undefined);
     const [overlayClicked, setOverlayClicked] = useState<boolean>(false);
-    const player = useRef<YoutubePlayerType>();
+    const [playerReady, setPlayerReady] = useState<boolean>(false);
 
-    const hasOverlay = overrideImage || posterImage;
+    const hasOverlay = !!(overrideImage || posterImage);
 
     /**
-     * Show the overlay if:
+     * Show an overlay if:
+     *
      * - It exists
      *
-     * and
+     * AND
      *
-     * - It hasn't been clicked upon
+     * - It hasn't been clicked
      */
     const showOverlay = hasOverlay && !overlayClicked;
+
     /**
      * Show a placeholder if:
      *
-     * - We don't have an iframe source yet (probably because we don't have consent)
-     *
-     * and
+     * - We don't have an overlay OR the user has clicked the overlay
      *
-     * - There's no overlay to replace it with or the reader clicked to play but we're
-     * still waiting on consent
+     * AND
      *
+     * - The player is not ready
      */
-    const showPlaceholder = !iframeSrc && (!hasOverlay || overlayClicked);
+    const showPlaceholder = (!hasOverlay || overlayClicked) && !playerReady;
```

Comment:
This is nice (using `playerReady` here). It makes sense having this wait controlled like this